### PR TITLE
show news details

### PR DESCRIPTION
**Pull request by Omid07**

### CODE REVIEW

File: app/src/main/java/com/example/omid/rss_5/NewsActivity.java [hunk 1]
```diff
@@ -35,7 +39,18 @@ protected void onCreate(Bundle savedInstanceState) {
 
     @Override
     public void setData(ArrayList<CategoryNews> categoryNewsList) {
+        for(int i=0;i<categoryNewsList.size();i++) {
+            mCategoryNewsesList.add(i,categoryNewsList.get(i));
+        }
         mAdapter = new CategoryNewsAdapter(categoryNewsList);
         mRecyclerView.setAdapter(mAdapter);
+
```
Comment (shawon-sharma) [inline app/src/main/java/com/example/omid/rss_5/NewsActivity.java]: remove blank line

File: app/src/main/java/com/example/omid/rss_5/adapter/CategoryNewsAdapter.java [hunk 2]
```diff
@@ -16,14 +17,17 @@
...
 public class CategoryNewsAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
     private ArrayList<CategoryNews> categoryNewsArrayList;
+    private Context mContext;
     public CategoryNewsAdapter(ArrayList<CategoryNews> categoryNewsList) {
         this.categoryNewsArrayList = categoryNewsList;
     }
 
     @Override
     public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+
```
Comment (shawon-sharma) [inline app/src/main/java/com/example/omid/rss_5/adapter/CategoryNewsAdapter.java]: remove blank

File: app/src/main/java/com/example/omid/rss_5/ui/CategoryNewsViewHolder.java [hunk 3]
```diff
@@ -1,29 +1,44 @@
...
+import com.example.omid.rss_5.NewsActivity;
 import com.example.omid.rss_5.R;
 import com.example.omid.rss_5.util.CategoryNews;
 
 /**
  * Created by omid on 6/3/16.
  */
-public class CategoryNewsViewHolder  extends RecyclerView.ViewHolder {
+public class CategoryNewsViewHolder  extends RecyclerView.ViewHolder implements View.OnClickListener  {
     private TextView textTitle;
```
Comment (shawon-sharma) [inline app/src/main/java/com/example/omid/rss_5/ui/CategoryNewsViewHolder.java]: naming convention

File: app/src/main/java/com/example/omid/rss_5/NewsActivity.java [hunk 4]
```diff
@@ -35,7 +39,18 @@ protected void onCreate(Bundle savedInstanceState) {
...
+    }
+    @Override
+    public void itemClicked(View view, int position) {
+        Intent intent = new Intent(NewsActivity.this, NewsDetailActivity.class);
+        CategoryNews categoryNews = mCategoryNewsesList.get(position);
+        intent.putExtra("CategoryNews", categoryNews);
```
Comment (shawon-sharma) [inline app/src/main/java/com/example/omid/rss_5/NewsActivity.java]: put in String file

File: app/src/main/java/com/example/omid/rss_5/ui/CategoryNewsViewHolder.java [hunk 5]
```diff
@@ -1,29 +1,45 @@
...
     private TextView textShortDescription;
     private TextView textDate;
-    public CategoryNewsViewHolder(View itemView) {
+    private ClickListener clickListener;
+    private Context context;
```
Comment (shawon-sharma) [inline app/src/main/java/com/example/omid/rss_5/ui/CategoryNewsViewHolder.java]: convention

File: app/src/main/res/layout/activity_news_detail.xml [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:paddingBottom="@dimen/activity_vertical_margin"
+        android:paddingLeft="@dimen/activity_horizontal_margin"
+        android:paddingRight="@dimen/activity_horizontal_margin"
+        android:paddingTop="@dimen/activity_vertical_margin"
+        tools:context="com.example.omid.rss_5.NewsDetailActivity">
+
+        <TextView
+            android:id="@+id/title"
```
Comment (shawon-sharma) [inline app/src/main/res/layout/activity_news_detail.xml]: naming convention i.e;(text_title)

File: app/src/main/res/layout/activity_news_detail.xml [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:textColor="@android:color/black"
+            android:textSize="@dimen/_20sp"
+            android:textStyle="bold"
+            android:text="asas"/>
```
Comment (shawon-sharma) [inline app/src/main/res/layout/activity_news_detail.xml]: remove text

File: app/src/main/java/com/example/omid/rss_5/ui/CategoryNewsViewHolder.java [hunk 8]
```diff
@@ -1,29 +1,45 @@
...
-    private TextView textShortDescription;
-    private TextView textDate;
-    public CategoryNewsViewHolder(View itemView) {
+public class CategoryNewsViewHolder  extends RecyclerView.ViewHolder implements View.OnClickListener  {
+    private TextView mTextTitle;
+    private TextView mTextShortDescription;
+    private TextView mTextDate;
+    private ClickListener mClickListener;
+    private Context mContext;
+    public CategoryNewsViewHolder(View itemView,Context context) {
```
Comment (ahammad-reyad) [inline app/src/main/java/com/example/omid/rss_5/ui/CategoryNewsViewHolder.java]: give space for better readability

File: app/src/main/java/com/example/omid/rss_5/ui/CategoryNewsViewHolder.java [hunk 9]
```diff
@@ -1,29 +1,45 @@
...
-    private TextView textTitle;
-    private TextView textShortDescription;
-    private TextView textDate;
-    public CategoryNewsViewHolder(View itemView) {
+public class CategoryNewsViewHolder  extends RecyclerView.ViewHolder implements View.OnClickListener  {
+    private TextView mTextTitle;
+    private TextView mTextShortDescription;
+    private TextView mTextDate;
+    private ClickListener mClickListener;
+    private Context mContext;
```
Comment (ahammad-reyad) [inline app/src/main/java/com/example/omid/rss_5/ui/CategoryNewsViewHolder.java]: put a extra line between variable and starting of method

Comment (ahammad-reyad) [inline app/src/main/java/com/example/omid/rss_5/ui/CategoryNewsViewHolder.java]: ok

Comment (ahammad-reyad) [inline app/src/main/java/com/example/omid/rss_5/ui/CategoryNewsViewHolder.java]: OK

**Comment by nvmanh** [comment]
OK!

File: app/src/main/java/com/example/omid/rss_5/NewsDetailActivity.java [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+        mPubDate = (TextView) findViewById(R.id.text_pub_date);
+        mImage = (ImageView) findViewById(R.id.image_news);
+        mLink = (TextView) findViewById(R.id.link);
+        mTitle.setText(categoryNews.getTitle());
+        mAuthor.setText(categoryNews.getAuthor());
+        mDescription.setText(categoryNews.getDescription());
+        mLink.setText(categoryNews.getLink());
+        mPubDate.setText(categoryNews.getPubDate());
+        Picasso.with(this).
+                load(categoryNews.getImage()).
```
Comment (nvmanh) [inline app/src/main/java/com/example/omid/rss_5/NewsDetailActivity.java]: should be

File: app/src/main/java/com/example/omid/rss_5/util/ProcessXML.java [hunk 11]
```diff
@@ -64,6 +64,17 @@ public ProcessXML(InputStream inputStream){
...
+                            Element element = (Element) currentChild;
+                            NamedNodeMap attributes = element.getAttributes();
+                            Node enclosureItem = attributes.getNamedItem("url");
+                            String image = enclosureItem.getNodeValue();
+                            mCategory.setImage(image);
+                            break;
+                        case Constant.LINK:
+                            String link = currentChild.getTextContent();
+                            mCategory.setLink(link);
+                            break;
```
Comment (nvmanh) [inline app/src/main/java/com/example/omid/rss_5/util/ProcessXML.java]: missing default case

**Comment by nvmanh** [comment]
OK!

**Pull request merged**